### DISCUSSION TITLE
update Patching.MaxStepsForScript programmatically

### DISCUSSION THREAD

**Discussion by mic2zar** [ravendb/ravendb#17673] [Q&A] [answered]
I am creating patches, which might require Patching.MaxStepsForScript to be set higher than its default value. I know how to update it in Studio (requires database restart), but I would like to include a routine in my code, which makes just a temporal change for the time, when the patch is getting executed and sets property to its original value afterwards. I could not really find out in the documentation or this forum if it is even possible. Also copilot generates quite useless code. Would you please let me know if it would be the right approach and if yes, how to achieve it. 

```
var record = await _store.Maintenance.Server.SendAsync(new GetDatabaseRecordOperation(_store.Database));
record.Settings["Patching.MaxStepsForScript"] = "100000"; 
await _store.Maintenance.Server.SendAsync(new UpdateDatabaseOperation(record));
```

**Comment by ppekrol** [comment, marked as answer]
This is fine, but you also need to reload the database for those settings to take effect. You can for example disable it and enable it, this will force the reload.

**Reply by mic2zar** [reply]
Thank you :). But UpdateDatabaseOperation requires some second parameter, which is `long etag`. I am not sure, what shall I submit - would some random number be enough. Or shall I obtain it from somewhere?

**Reply by ppekrol** [reply]
As a result of GetDatabaseRecordOperation you are getting 'DatabaseRecordWithEtag', use the Etag value from there.

**Reply by mic2zar** [reply]
In case somebody would be searching for an example I leave here the working one:

```
var recordOperation = new GetDatabaseRecordOperation(_store.Database);
var record = await _store.Maintenance.Server.SendAsync(recordOperation);
record.Settings["Patching.MaxStepsForScript"] = "1000000"; 
await _store.Maintenance.Server.SendAsync(new UpdateDatabaseOperation(record, record.Etag));
```

What I am still missing is the possibility to restart database from the API. Apparently new Settings will not take effect till database is restarted, which I currently can only perform from the Studio. Are you planning to add database restart to API?

**Reply by garayx** [reply]
You can use the  `ToggleDatabasesStateOperation`
https://ravendb.net/docs/article-page/5.4/csharp/client-api/operations/server-wide/toggle-databases-state

**Comment by lahma** [comment]
We run `"Patching.MaxStepsForScript": 0` in production as we control the queries (we write them). RavenDB has a conservative default, but we haven't run into a situation where we would like query to fail when we need the results.
